scripts/pythondistdeps.py

```diff
@@ -11,21 +11,79 @@
 # RPM python dependency generator, using .egg-info/.egg-link/.dist-info data
 #
 
-# Please know:
-# - Notes from an attempted rewrite from pkg_resources to importlib.metadata in
-#   2020 can be found in the message of the commit that added this line.
-
 from __future__ import print_function
 import argparse
-from os.path import basename, dirname, isdir, sep
-from sys import argv, stdin, version
 from distutils.sysconfig import get_python_lib
+from itertools import chain
+from os.path import dirname, sep
+from packaging.requirements import Requirement
+from packaging.version import parse
+import re
+from sys import argv, stdin
 from warnings import warn
 
+try:
+    from importlib.metadata import PathDistribution
+except ImportError:
+    from importlib_metadata import PathDistribution
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+
+
+class Req(Requirement):
+    def __init__(self, requirement_string):
+        super(Req, self).__init__(requirement_string)
+        self.normalized_name = normalize_name(self.name)
+        self.legacy_normalized_name = legacy_normalize_name(self.name)
```

Comment:
It's confusing for me to not see the functions defined before this class.

Reply:
I'm trying to keep the diff small, but I can move the function definition up if you wish.

Reply:
With a change this big, I wouldn't worry about that.